src/basic/os-util.c

```diff
@@ -249,35 +271,35 @@ int fopen_extension_release(const char *root, const char *extension, bool relax_
         return 0;
 }
 
-static int parse_release_internal(const char *root, bool relax_extension_release_check, const char *extension, va_list ap) {
+static int parse_release_internal(const char *root, ImageClass image_class, bool relax_extension_release_check, const char *extension, va_list ap) {
         _cleanup_fclose_ FILE *f = NULL;
         _cleanup_free_ char *p = NULL;
         int r;
 
-        r = fopen_extension_release(root, extension, relax_extension_release_check, &p, &f);
+        r = fopen_extension_release(root, image_class, extension, relax_extension_release_check, &p, &f);
         if (r < 0)
                 return r;
 
         return parse_env_filev(f, p, ap);
 }
 
-int _parse_extension_release(const char *root, bool relax_extension_release_check, const char *extension, ...) {
+int _parse_extension_release(const char *root, ImageClass image_class, bool relax_extension_release_check, const char *extension, ...) {
         va_list ap;
         int r;
 
         va_start(ap, extension);
-        r = parse_release_internal(root, relax_extension_release_check, extension, ap);
+        r = parse_release_internal(root, image_class, relax_extension_release_check, extension, ap);
         va_end(ap);
 
         return r;
 }
 
-int _parse_os_release(const char *root, ...) {
+int _parse_os_release(const char *root, ImageClass image_class,...) {
```

Comment:
parse_os_release should not take an image class, it's never about extensions, it's about the host

Reply:
But some of the env variables within that file depend on the image class right, like whether to look for SYSCFG_LEVEL or SYSEXT_LEVEL and scope similarly...

Reply:
The sysext image class information needs to be accessed by the os-util file somehow for some of its functions, what's a good alternative to do that if not passing it in this function. This is the only function that is called of os-util from sysext.c

Reply:
Just pass -EINVAL in those cases